jerry-core/ecma/base/ecma-globals.h

```diff
@@ -214,84 +204,78 @@ typedef enum
 } ecma_property_configurable_value_t;
 
 /**
- * Width of internal property type field's width
+ * Property's flag list.
  */
-#define ECMA_PROPERTY_INTERNAL_PROPERTY_TYPE_WIDTH   (5)
+typedef enum
+{
+  ECMA_PROPERTY_FLAG_NAMEDDATA = 1u << 0, /**< property is named data */
+  ECMA_PROPERTY_FLAG_NAMEDACCESSOR = 1u << 1, /**< property is named accessor */
+  ECMA_PROPERTY_FLAG_INTERNAL = 1u << 2, /**< property is internal property */
+  ECMA_PROPERTY_FLAG_CONFIGURABLE = 1u << 3, /**< property is configurable */
+  ECMA_PROPERTY_FLAG_ENUMERABLE = 1u << 4, /**< property is enumerable */
+  ECMA_PROPERTY_FLAG_WRITABLE = 1u << 5, /**< property is writable */
+  ECMA_PROPERTY_FLAG_LCACHED = 1u << 6, /**< property is lcached */
+} ecma_property_flags_t;
 
 /**
  * Pair of pointers - to property's getter and setter
  */
 typedef struct
 {
-  __extension__ mem_cpointer_t getter_p : ECMA_POINTER_FIELD_WIDTH; /**< pointer to getter object */
-  __extension__ mem_cpointer_t setter_p : ECMA_POINTER_FIELD_WIDTH; /**< pointer to setter object */
+  mem_cpointer_t getter_p; /**< pointer to getter object */
+  mem_cpointer_t setter_p; /**< pointer to setter object */
 } ecma_getter_setter_pointers_t;
 
 /**
  * Description of ecma-property
  */
-typedef struct __attr_packed___ ecma_property_t
+typedef struct ecma_property_t
 {
-  /** Property's type (ecma_property_type_t) */
-  unsigned int type : 2;
-
   /** Compressed pointer to next property */
-  __extension__ mem_cpointer_t next_property_p : ECMA_POINTER_FIELD_WIDTH;
+  mem_cpointer_t next_property_p;
+
+  /** Property's flags (ecma_property_flags_t) */
+  uint8_t flags;
 
-  /** Property's details (depending on Type) */
+  /** Property's header part (depending on Type) */
   union
   {
-    /** Description of named data property */
-    struct __attr_packed___ ecma_named_data_property_t
-    {
-      /** Value */
-      __extension__ ecma_value_t value : ECMA_VALUE_SIZE;
+    /** Named data property value upper bits */
+    uint8_t named_data_property_value_high;
+    /** Internal property type */
+    uint8_t internal_property_type;
+  } h;
```

Comment:
The problem is the names will be too long, and in some cases it is easy to reach the 120 char.